arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/Sequence.kt

```diff
@@ -3,6 +3,202 @@ package arrow.core
 import arrow.typeclasses.Monoid
 import arrow.typeclasses.Semigroup
 
+fun <B, C, D> mapN(
+  b: Sequence<B>,
+  c: Sequence<C>,
+  map: (B, C) -> D
+): Sequence<D> =
+  mapN(
+    b,
+    c,
+    sequenceOf(Unit),
```

Comment:
Does `sequence(Unit)` allocate new sequences or is it backed by a singleton value? If not we could optimize this.

Reply:
It's probably not backed by a singleton value.